geoutils/georaster/raster.py

```diff
@@ -846,7 +835,24 @@ def data(self, new_data: np.ndarray | np.ma.masked_array) -> None:
                 f"New data must be of the same shape as existing data: {orig_shape}. Given: {new_data.shape[1:]}."
             )
 
-        self._data = np.ma.masked_array(new_data, fill_value=self.nodata)
+        # If the new data is not masked and has non-finite values, we define a fill_value
+        if (not np.ma.is_masked(new_data) and self.nodata is None and np.count_nonzero(~np.isfinite(new_data)) > 0) or (
+            np.ma.is_masked(new_data)
+            and self.nodata is None
+            and np.count_nonzero(~np.isfinite(new_data.data[~new_data.mask])) > 0
+        ):
+            self._nodata = _default_ndv(dtype)
```

Comment:
Maybe we should raise a warning in case the default value is used in the array.

Reply:
Done + test.